stampy/plugin/stock.py

```diff
@@ -54,11 +54,15 @@ def __init__(self):
 
     def get(self, symbol):
         url = self.prefix + "%s/quote" % symbol
-        content = requests.get(url).content
+        content = requests.get(url).content.split(",")
         quote = {'t': symbol}
-        quote['l_cur'] = content.split(",")[9].split(":")[1]
-        quote['c'] = content.split(",")[20].split(":")[1]
-        quote['cp'] = float(content.split(",")[21].split(":")[1]) * 100
+        for x in content:
```

Comment:
what about:

~~~py
def get(self, symbol):
  url = self.prefix + "%s/quote" % symbol
  content = json.loads(requests.get(url).content)
  quote = {'t': symbol}

  if "change" in content:
   quote['c']=content['change']
  if "changePercent" in content:
   quote['cp']=content['changePercent']
  if "latestPrice" in content:
    quote['l_cur'] = content['latestPrice']
  return quote
~~~


using json.loads, converts the content into a dictionary (parsing the return of the api), so it returns something like:

{u'avgTotalVolume': 1837021,
  u'calculationPrice': u'close',
  u'change': 0.23,
  u'changePercent': 0.0019,
  u'close': 121.12,
  u'closeTime': 1508270652819,
  u'companyName': u'Red Hat Inc.',
  u'delayedPrice': 121.12,
  u'delayedPriceTime': 1508273890029,
  u'iexAskPrice': 0,
  u'iexAskSize': 0,
  u'iexBidPrice': 0,
  u'iexBidSize': 0,
  u'iexLastUpdated': 1508270387285,
  u'iexMarketPercent': 0.04233,
  u'iexRealtimePrice': 121.07,
  u'iexRealtimeSize': 100,
  u'iexVolume': 46128,
  u'latestPrice': 121.12,
  u'latestSource': u'Close',
  u'latestTime': u'October 17, 2017',
  u'latestUpdate': 1508270652819,
  u'latestVolume': 1089724,
  u'marketCap': 21431842805,
  u'open': 120.36,
  u'openTime': 1508247000757,
  u'peRatio': 81.84,
  u'previousClose': 120.89,
  u'primaryExchange': u'New York Stock Exchange',
  u'sector': u'Technology',
  u'symbol': u'RHT',
  u'week52High': 122.31,
  u'week52Low': 68.54,
  u'ytdChange': 0.7273245864232744}